Data/models/DataController.swift

```diff
@@ -30,22 +31,75 @@ public class DataController: NSObject {
-            return try FileManager.default.contentsOfDirectory(at: url, includingPropertiesForKeys: nil, options: []).filter({$0.lastPathComponent.hasPrefix(dbName)})
+    public func migrateToNewPathIfNeeded() throws {
+        enum MigrationError: Error {
+            case OldStoreMissing(String)
+            case MigrationFailed(String)
+            case CleanupFailed(String)
         }
-        if FileManager.default.fileExists(atPath: oldStoreURL.path) && !storeExists() {
-            do {
-                try migrationContainer.persistentStoreCoordinator.addPersistentStore(ofType: NSSQLiteStoreType, configurationName: nil, at: oldStoreURL, options: nil)
-                if let oldStore = migrationContainer.persistentStoreCoordinator.persistentStore(for: oldStoreURL) {
-                    try migrationContainer.persistentStoreCoordinator.migratePersistentStore(oldStore, to: storeURL, options: nil, withType: NSSQLiteStoreType)
-                    try migrationContainer.persistentStoreCoordinator.destroyPersistentStore(at: oldStoreURL, ofType: NSSQLiteStoreType, options: nil)
-                    try sqliteFiles(from: oldStoreURL.deletingLastPathComponent(), dbName: DataController.databaseName).forEach(FileManager.default.removeItem)
-                }
-            } catch {
-                log.error(error)
-            }
+        
+        // This logic must account for 4 different situations:
+        // 1. New Users (no migration, use new location
+        // 2. Upgraded users with successful migration (use new database)
+        // 3. Upgraded users with unsuccessful migrations (use new database, ignore old files)
+        // 4. Upgrading users (attempt migration, if fail, use old store, if successful delete old files)
+        //      - re-attempt migration on every new app version, until they are in #2
+        
+        let specificStoreExists: (URL) -> Bool = { url in
+            FileManager.default.fileExists(atPath: url.path)
+        }
+        
+        if !specificStoreExists(oldDocumentStoreURL) || specificStoreExists(supportStoreURL) {
+            // Old store absent, no data to migrate (#1 | #3)
+            // or
+            // New store already exists, do not attempt to overwrite (#2)
+            
+            // Update flag to avoid re-running this logic
+            Preferences.Database.DocumentToSupportDirectoryMigration.completed.value = true
+            return
+        }
+        
+        // Going to attempt migration (#4 in some level)
+        
+        let coordinator = migrationContainer.persistentStoreCoordinator
+
+        guard let oldStore = coordinator.persistentStore(for: oldDocumentStoreURL) else {
+            throw MigrationError.OldStoreMissing("Old store unavailable")
+        }
+        
+        // Attempting actual database migration Document -> Support 🤞
+        do {
+            let migrationOptions = [
+                NSPersistentStoreFileProtectionKey: true
```

Comment:
Is this the key that will most likely help with this migration bug in 1.12?
If so I would add a comment here

Reply:
I don't think it is actually necessary. The modification to `addPersistentStore` should be enough. I just added this in case it was useful for adding file protection for the new store. @Brandon-T is submitting a new PR to handle some of the encryption concerns too at some point.